onadata/apps/api/viewsets/xform_list_api.py

```diff
@@ -86,25 +87,11 @@ def filter_queryset(self, queryset):
                 # including those shared by other users
                 queryset = super().filter_queryset(queryset)
         else:
-            profile = get_object_or_404(
-                UserProfile, user__username=username.lower()
+            # Only returns non-secured projects with URL starts with username
+            queryset = queryset.filter(
+                user__username=username.lower(), require_auth=False
```

Comment:
Comment could use a little work: "returns" would be better as "return" (imperative), and "with" should perhaps be "when"?

I don't know about introducing the terminology "secured". My preference would be to stick with something longer but more consistent with what we already have, like "Only return projects that allow anonymous submissions when path starts with a username"